config/src/inline/natspec.rs

```diff
@@ -0,0 +1,207 @@
+                if let Some(node) = contract_root_node(&ast.nodes, &contract) {
+                    apply(&mut natspecs, &contract, node)
+                }
+            }
+        }
+        natspecs
+    }
+
+    /// Returns a string describing the natspec
+    /// context, for debugging purposes 🐞 <br>
+    /// i.e. `dir/TestContract.t.sol:FuzzContract:test_myFunction:10:12:111`
+    pub fn debug_context(&self) -> String {
+        format!("{}:{}:{}", self.contract, self.function, self.line)
+    }
+
+    /// Returns a list of configuration lines that match the current profile
+    pub fn current_profile_configs(&self) -> Vec<String> {
+        let prefix: &str = INLINE_CONFIG_PREFIX_SELECTED_PROFILE.as_ref();
+        self.config_lines_with_prefix(prefix)
+    }
+
+    /// Returns a list of configuration lines that match a specific string prefix
+    pub fn config_lines_with_prefix<S: Into<String>>(&self, prefix: S) -> Vec<String> {
+        let prefix: String = prefix.into();
+        self.config_lines().into_iter().filter(|l| l.starts_with(&prefix)).collect()
+    }
+
+    /// Returns a list of all the configuration lines available in the natspec
+    pub fn config_lines(&self) -> Vec<String> {
+        self.docs
+            .split('\n')
+            .map(remove_whitespaces)
+            .filter(|line| line.contains(INLINE_CONFIG_PREFIX))
+            .collect::<Vec<String>>()
+    }
+}
+
+/// Given a list of nodes, find a "ContractDefinition" node that matches
+/// the provided contract_id.
+fn contract_root_node<'a>(nodes: &'a [Node], contract_id: &'a str) -> Option<&'a Node> {
+    for n in nodes.iter() {
+        if let NodeType::ContractDefinition = n.node_type {
+            let contract_data = &n.other;
+            if let Value::String(contract_name) = contract_data.get("name")? {
+                if contract_id.ends_with(contract_name) {
+                    return Some(n)
+                }
+            }
+        }
+    }
+    None
+}
+
+/// Implements a DFS over a compiler output node and its children.
+/// If a natspec is found it is added to `natspecs`
+fn apply(natspecs: &mut Vec<NatSpec>, contract: &str, node: &Node) {
+    for n in node.nodes.iter() {
+        if let Some((function, docs, line)) = get_fn_data(n) {
+            natspecs.push(NatSpec { contract: contract.into(), function, line, docs })
+        }
+        apply(natspecs, contract, n);
+    }
+}
+
+fn get_fn_data(node: &Node) -> Option<(String, String, String)> {
+    if let NodeType::FunctionDefinition = node.node_type {
+        let fn_data = &node.other;
+        let fn_name: String = get_fn_name(fn_data)?;
+        let (fn_docs, docs_src_line): (String, String) = get_fn_docs(fn_data)?;
+        return Some((fn_name, fn_docs, docs_src_line))
+    }
+
+    None
+}
+
+fn get_fn_name(fn_data: &BTreeMap<String, Value>) -> Option<String> {
+    match fn_data.get("name")? {
+        Value::String(fn_name) => Some(fn_name.into()),
+        _ => None,
+    }
+}
```

Comment:
Could we add some docs for these functions, even if they're only used internally?